diff -r -u a/src/terminal-util.c b/src/terminal-util.c
--- a/src/terminal-util.c	2018-12-05 19:47:58.000000000 +0200
+++ b/src/terminal-util.c	2018-12-05 19:32:09.302636807 +0200
@@ -111,7 +111,7 @@
     }
 }
 
-static gboolean
+gboolean
 open_url (GtkWindow *parent,
           const char *uri,
           guint32 user_time,
diff -r -u a/src/terminal-util.h b/src/terminal-util.h
--- a/src/terminal-util.h	2018-12-05 19:47:58.000000000 +0200
+++ b/src/terminal-util.h	2018-12-05 19:32:30.015120734 +0200
@@ -110,6 +110,12 @@
 void terminal_util_save_print_settings (GtkPrintSettings *settings,
                                         GtkPageSetup *page_setup);
 
+gboolean
+open_url (GtkWindow *parent,
+          const char *uri,
+          guint32 user_time,
+          GError **error);
+
 G_END_DECLS
 
 #endif /* TERMINAL_UTIL_H */
diff -r -u a/src/terminal-window.c b/src/terminal-window.c
--- a/src/terminal-window.c	2018-12-05 19:47:58.000000000 +0200
+++ b/src/terminal-window.c	2018-12-05 19:37:43.101159703 +0200
@@ -654,6 +654,29 @@
   vte_terminal_copy_clipboard_format (VTE_TERMINAL (priv->active_screen), format);
 }
 
+static void
+action_google_cb (GSimpleAction *action,
+                GVariant *parameter,
+                gpointer user_data)
+{
+  TerminalWindow *window = user_data;
+  TerminalWindowPrivate *priv = window->priv;
+  const char *format_str;
+  VteFormat format;
+  gchar *url;
+  GtkClipboard *clipboard;
+
+  if (priv->active_screen == NULL)
+    return;
+
+  g_assert_nonnull (parameter);
+  g_variant_get (parameter, "&s", &format_str);
+  clipboard = gtk_widget_get_clipboard (GTK_WIDGET (window), GDK_SELECTION_PRIMARY);
+  url = g_strdup_printf ("https://www.google.com/search?ie=UTF-8&q=%s", gtk_clipboard_wait_for_text(clipboard));
+  open_url (GTK_WINDOW (window), url, gtk_get_current_event_time (), NULL);
+  g_free(url);
+}
+
 /* Clipboard helpers */
 
 typedef struct {
@@ -1781,6 +1804,7 @@
   g_menu_append (section4, _("_Copy"), "win.copy::text");
   g_menu_append (section4, _("Copy as _HTML"), "win.copy::html");
   g_menu_append (section4, _("_Paste"), "win.paste-text");
+  g_menu_append (section4, _("Search on Google"), "win.search-google::text");
   if (g_action_get_enabled (G_ACTION (lookup_action (window, "paste-uris"))))
     g_menu_append (section4, _("Paste as _Filenames"), "win.paste-uris");
 
@@ -2119,6 +2143,7 @@
     { "paste-uris",          action_paste_uris_cb,       NULL,   NULL, NULL },
     { "reset",               action_reset_cb,            "b",    NULL, NULL },
     { "select-all",          action_select_all_cb,       NULL,   NULL, NULL },
+    { "search-google",       action_google_cb,           "s",    NULL, NULL },
     { "size-to",             action_size_to_cb,          "(uu)", NULL, NULL },
     { "tab-detach",          action_tab_detach_cb,       NULL,   NULL, NULL },
     { "tab-move-left",       action_tab_move_left_cb,    NULL,   NULL, NULL },
